File: JULY-2025/22-07-2025/p1695.dart
class Solution {
  int maximumUniqueSubarray(List<int> nums) {
    Set<int> seen = Set<int>();
    int maxSum = 0;
    int currentSum = 0;
    int left = 0;
    for (int right = 0; right < nums.length; right++) {
        while (seen.contains(nums[right])) {
            seen.remove(nums[left]);
            currentSum -= nums[left];
            left++;
        }
        seen.add(nums[right]);
        currentSum += nums[right];
        if (currentSum > maxSum) {
            maxSum = currentSum;
        }
    }
    return maxSum;
    }
}

File: JULY-2025/22-07-2025/p1695.swift
class Solution {
    func maximumUniqueSubarray(_ nums: [Int]) -> Int {
        let n = nums.count
        var seen = Set<Int>()
        seen.reserveCapacity(n)
        
        var left = 0
        var currentSum = 0
        var maxSum = 0
        
        for (right, value) in nums.enumerated() {
            while seen.contains(value) {
                currentSum -= nums[left]
                seen.remove(nums[left])
                left += 1
            }
            currentSum += nums[right]
            seen.insert(value)
            maxSum = max(maxSum, currentSum)
        }
        
        return maxSum        
    }
}